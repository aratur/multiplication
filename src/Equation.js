import React from 'react';
import PropTypes from 'prop-types';
import EquationSignButton from './EquationSignButton';
import EquationNumberButton from './EquationNumberButton';

 const Equation = (props) => {
  return (
    <div className="jumbotron" style={({margin: "10px", padding: "10px"})}>
      <EquationNumberButton value={props.xValue} />
      <EquationSignButton value="X" />
      <EquationNumberButton value={props.yValue} />
      <EquationSignButton value="=" />
      <EquationNumberButton value={props.correctAnswer} />
    </div>
  );
};

Equation.propTypes = {
   xValue: PropTypes.number.isRequired,
   yValue: PropTypes.number.isRequired,
   correctAnswer: PropTypes.string.isRequired
}

export default Equation;
