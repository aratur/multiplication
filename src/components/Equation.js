/* eslint "react/button-has-type": "off" */
import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import multiply from '../img/multiply.png';
import equals from '../img/equals.png';
import { i18n, getTranslations } from '../redux-store/i18nSlice';

// without JSX
const Equation = (props) => {
  const numberProps = {
    style: {
      margin: '5px',
      width: '55px',
      height: '55px',
      padding: '0px',
    },
    type: 'button',
    tabIndex: '-1',
    role: 'option',
    className: 'btn btn-success btn-lg',
  };
  const signProps = {
    style: {
      margin: '5px',
      width: '55px',
      height: '55px',
      padding: '12px',
    },
  };
  const translations = useSelector(getTranslations);
  const { xValue, yValue, correctAnswer } = props;

  return React.createElement(
    'div',
    { className: 'well text-center', style: { padding: '10px' } },
    React.createElement(
      'label',
      {
        htmlFor: 'equation',
      },
      i18n(translations, 'equation.label'),
      React.createElement(
        'div',
        { id: 'equation' },
        React.createElement(
          'button',
          { name: `number ${xValue}`, ...numberProps },
          xValue
        ),
        React.createElement('img', {
          src: multiply,
          alt: 'multiply',
          ...signProps,
        }),
        React.createElement(
          'button',
          { name: `by ${yValue}`, ...numberProps },
          yValue
        ),
        React.createElement('img', {
          src: equals,
          alt: 'equals',
          ...signProps,
        }),
        React.createElement(
          'button',
          { name: `answer ${correctAnswer}`, ...numberProps },
          correctAnswer
        )
      )
    )
  );
};

Equation.propTypes = {
  xValue: PropTypes.number.isRequired,
  yValue: PropTypes.number.isRequired,
  correctAnswer: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
    .isRequired,
};

export default Equation;
