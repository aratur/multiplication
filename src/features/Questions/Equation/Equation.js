import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import multiply from '../../../assets/multiply.png';
import equals from '../../../assets/equals.png';
import { i18n, getTranslations } from '../../../redux-store/i18nSlice';
import {
  getXValue,
  getYValue,
  getAnswer,
  getStatus,
  setNext,
} from '../../../redux-store/questionSlice';

// without JSX
const Equation = () => {
  const numberProps = {
    style: {
      marginTop: '5px',
      width: '50px',
      height: '50px',
    },
    type: 'button',
    tabIndex: '-1',
    role: 'option',
    className: 'btn btn-success btn-lg',
  };
  const signProps = {
    style: {
      marginTop: '5px',
      width: '50px',
      height: '50px',
      padding: '10px',
    },
    className: 'img-rounded',
  };
  const translations = useSelector(getTranslations);
  const dispatch = useDispatch();
  const xValue = useSelector(getXValue);
  const yValue = useSelector(getYValue);
  const answer = useSelector(getAnswer);
  const status = useSelector(getStatus);
  const correctAnswer = answer === -1 ? '?' : xValue * yValue;

  useEffect(() => {
    if (status === 'idle') {
      dispatch(setNext());
    }
  }, [dispatch, status]);

  return React.createElement(
    'div',
    { className: 'well text-center', style: { padding: '10px' } },
    React.createElement(
      'label',
      {
        htmlFor: 'equation',
      },
      i18n(translations, 'equation.label'),
      React.createElement(
        'div',
        { id: 'equation' },
        React.createElement(
          'button',
          {
            name: `number ${xValue}`,
            'data-testid': 'question-left',
            type: 'button',
            ...numberProps,
          },
          xValue
        ),
        React.createElement('img', {
          src: multiply,
          alt: 'multiply',
          ...signProps,
        }),
        React.createElement(
          'button',
          {
            name: `by ${yValue}`,
            'data-testid': 'question-right',
            type: 'button',
            ...numberProps,
          },
          yValue
        ),
        React.createElement('img', {
          src: equals,
          alt: 'equals',
          ...signProps,
        }),
        React.createElement(
          'button',
          { name: `answer ${correctAnswer}`, type: 'button', ...numberProps },
          correctAnswer
        )
      )
    )
  );
};

export default Equation;
