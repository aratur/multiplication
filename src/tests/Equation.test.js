import React from 'react';
import { render } from 'react-dom';
import pretty from 'pretty';
import Equation from '../components/Equation';

describe('Equation', () => {
  it('should render correctly', () => {
    const container = document.createElement('div');
    document.body.appendChild(container);
    render(
      <Equation
        xValue={2}
        yValue={2}
        correctAnswer={22}
      />, container,
    );
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
