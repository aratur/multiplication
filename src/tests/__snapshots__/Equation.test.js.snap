// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Equation should render correctly 1`] = `"<div class=\\"well\\" style=\\"padding: 10px;\\"><label for=\\"equation\\">Przeczytaj r√≥wnanie<div id=\\"equation\\"><button name=\\"number 2\\" style=\\"margin: 5px; width: 55px; height: 55px; padding: 0px;\\" type=\\"button\\" tabindex=\\"-1\\" role=\\"option\\" class=\\"btn btn-success btn-lg\\">2</button><img src=\\"multiply.png\\" alt=\\"multiply\\" style=\\"margin: 5px; width: 55px; height: 55px; padding: 12px;\\"><button name=\\"by 2\\" style=\\"margin: 5px; width: 55px; height: 55px; padding: 0px;\\" type=\\"button\\" tabindex=\\"-1\\" role=\\"option\\" class=\\"btn btn-success btn-lg\\">2</button><img src=\\"equals.png\\" alt=\\"equals\\" style=\\"margin: 5px; width: 55px; height: 55px; padding: 12px;\\"><button name=\\"answer 22\\" style=\\"margin: 5px; width: 55px; height: 55px; padding: 0px;\\" type=\\"button\\" tabindex=\\"-1\\" role=\\"option\\" class=\\"btn btn-success btn-lg\\">22</button></div></label></div>"`;
